packages/experience/src/containers/VerificationCode/index.tsx

```diff
@@ -42,15 +45,27 @@ const VerificationCode = ({ flow, identifier, className, hasPasswordButton, targ
     target
   );
 
-  useEffect(() => {
-    if (code.length === defaultLength && code.every(Boolean)) {
-      const payload =
-        identifier === SignInIdentifier.Email
-          ? { email: target, verificationCode: code.join('') }
-          : { phone: target, verificationCode: code.join('') };
-      void onSubmit(payload);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleSubmit = useCallback(async () => {
+    if (!isCodeReady) {
+      return;
     }
-  }, [code, identifier, onSubmit, target]);
+
+    setIsSubmitting(true);
+
+    await onSubmit(
+      identifier === SignInIdentifier.Email
+        ? { email: target, verificationCode: code.join('') }
+        : { phone: target, verificationCode: code.join('') }
+    );
+
+    setIsSubmitting(false);
```

Comment:
what if onSubmit throws error?

Reply:
onSubmit won't throw, all errors are handled in side the `onSubmit`.